docs/language/informal/covariant-overrides.md

```diff
@@ -0,0 +1,742 @@
+can be implemented, and both choices provide a certain amount of protection
+against tightening types by accident. We have chosen the permissive variant,
+and it is always possible to make a linter require a stricter one.
+
+For a regular (non-covariant) parameter, the override rule is is unchanged:
+its type must be a supertype of the type declared for the same parameter in
+each directly overridden declaration. This includes the typical case where the
+type does not change, because any type is a supertype of itself.
+
+For a covariant parameter, the override rule is that its type must be either a
+supertype or a subtype of the type declared for the same parameter in each
+of the directly or indirectly overridden declarations.
+
+It is not enough to require a relationship with the directly overridden
+declarations: If we only required a subtype or supertype relation to each
+directly overridden declaration, we can easily create an example showing
+that there are no guarantees about the relationship between the statically
+known (original) parameter type and the actual (desired) parameter type
+at run time:
+
+```dart
+class A { void foo(int i) {...}}
+class B implements A { void foo(Object o) {...}}
+class C implements B { void foo(covariant String s) {...}}
+
+main() {
+  A a = new C();
+  a.foo(42); // Original type: int, desired type: String: Unrelated.
+}
+```
+
+Checking all overridden declarations may seem expensive, because all
+supertypes must be inspected in order to find those declarations. However, we
+expect that the cost can be kept at a reasonable level:
+
+First, covariant parameters are expected to be rare, and it is possible for
+a compiler to detect them at a reasonable cost: For each parameter, the status
+of being covariant or not covariant can be maintained by storing it after
+inspecting the stored value for that parameter in each directly overridden
+method. With that information available, it is cheap to detect whether a given
+method has any covariant parameters. If that is not the case then the override
+checks are unchanged, compared to the language without this feature.
+
+Otherwise the parameter is covariant, and in this case it will be necessary to
+find all overridden declarations (direct and indirect) for that method, and
+gather all types for that parameter. Given that overridden methods would
+themselves often have a corresponding covariant parameter, it may be worthwhile
+to cache the result. In that case there will be a set of types that occur as
+the declared type of the given parameter, and the check will then be to iterate
+over these overridden parameter types and verify that the overriding parameter
+type is a subtype or a supertype of each of them.
+
+#### Subtyping
+
+`covariant` modifiers are ignored when deciding subtype relationships among
+classes. So this:
+
+```dart
+var group = new RadioGroup();
+Widget widget = group;
+```
+
+... is perfectly fine.
```

Comment:
I'm not sure what this shows - subtyping never had anything to do with member signatures. Either elaborate what the concern could be, or remove the section, because I can't what point it's trying to make.